src/go/metrics/sender.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2021 The Magma Authors.
+//
+// This source code is licensed under the BSD-style license found in the
+// LICENSE file in the root directory of this source tree.
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrics
+
+import (
+	"context"
+
+	io_prometheus_client "github.com/prometheus/client_model/go"
+
+	"github.com/go-openapi/swag"
+	"github.com/magma/magma/protos/magma/metricsd"
+	"github.com/pkg/errors"
+)
+
+type MetricSender interface {
+	// ProcessGauge
+	ProcessGauge(gauge Gauge)
+
+	// Send sends the processed metrics to the destination
+	Send() error
+}
+
+//go:generate go run github.com/golang/mock/mockgen -destination mock_sender/mock_sender.go . MetricSender
+
+type prometheusMetricSender struct {
+	metricsContainer *metricsd.MetricsContainer
+	gatewayId        string
+	client           metricsd.MetricsControllerClient
+}
+
+func NewPrometheusMetricSender(client metricsd.MetricsControllerClient, gatewayId string) MetricSender {
+	// return &prometheusMetricSender{&metricsd.MetricsContainer{}}
+	// container := &metricsd.MetricsContainer{GatewayId: "asdf"}
+	container := &metricsd.MetricsContainer{
+		GatewayId: gatewayId,
+		Family:    []*io_prometheus_client.MetricFamily{},
```

Comment:
dang, it would be nice if `metricsd.MetricsContainer` didn't use `io_prometheus_client.MetricFamily` 

not worth addressing at the moment, and this new `metrics` package should make it so that no one outside of `metrics` should need to use `io_prometheus_client`

Reply:
Still planning on removing this from the `metrics` package, but the Orc8r side needs to change first - I think I put some notes about this somewhere in the PR.